test/parallel/test-dgram-multicast-set-interface-lo.js

```diff
@@ -0,0 +1,275 @@
+
+              assert.equal(count, worker.messagesNeeded.length,
+                           'A worker received an invalid multicast message');
+            });
+
+            clearTimeout(timer);
+            console.error('[PARENT] Success');
+            killChildren(workers);
+          }
+        }
+      });
+    })(x);
+  }
+
+  var sendSocket = dgram.createSocket({
+    type: UDP[FAM],
+    reuseAddr: true
+  });
+
+  // don't bind the address explicitly for sending
+  sendSocket.bind(common.PORT, ANY[FAM]);
+  sendSocket.on('listening', function() {
+    // but explicitly set the outgoing interface
+    sendSocket.setMulticastInterface(interfaceAddress);
+    console.error(`outgoing iface ${interfaceAddress}`);
+  });
+
+  sendSocket.on('close', function() {
+    console.error('[PARENT] sendSocket closed');
+  });
+
+  sendSocket.sendNext = function() {
+    const msg = messages[i++];
+
+    if (!msg) {
+      try { sendSocket.close(); } catch (e) {}
+      return;
+    }
+    console.error(TMPL(NOW, msg.tail));
+    const buf = Buffer.from(TMPL(msg.tail));
+    if (msg.newAddr) {
+      console.error(`changing outgoing multicast ${msg.newAddr}`);
+      sendSocket.setMulticastInterface(msg.newAddr);
+    }
+
+    sendSocket.send(
+      buf,
+      0,
+      buf.length,
+      common.PORT,
+      msg.mcast,
+      function(err) {
+        if (err) throw err;
+        console.error('[PARENT] sent %s to %s:%s',
+                      util.inspect(buf.toString()),
+                      msg.mcast, common.PORT);
+
+        process.nextTick(sendSocket.sendNext);
+      }
+    );
+  };
+
+  function killChildren(children) {
+    Object.keys(children).forEach(function(key) {
+      var child = children[key];
+      child.kill();
+    });
+  }
+}
+
+if (process.argv[2] === 'child') {
+  const IFACE = process.argv[3];
+  const MULTICAST = process.argv[4];
+  const NEEDEDMSGS = Number(process.argv[5]);
+  const SESSION = Number(process.argv[6]);
+  const receivedMessages = [];
+
+  console.error(`pid ${process.pid} iface ${IFACE} MULTICAST ${MULTICAST}`);
+  var listenSocket = dgram.createSocket({
+    type: UDP[FAM],
+    reuseAddr: true
+  });
+
+  listenSocket.on('message', function(buf, rinfo) {
+    // receive udp messages only sent from parent
+    if (!buf.toString().startsWith(SESSION)) return;
+
+    console.error('[CHILD] %s received %s from %j',
+                  process.pid,
+                  util.inspect(buf.toString()),
+                  rinfo);
+
+    receivedMessages.push(buf);
+
+    process.send({message: buf.toString()});
+
+    if (receivedMessages.length == NEEDEDMSGS) {
+      process.nextTick(function() {
+        listenSocket.close();
+      });
+    }
+  });
+
+  listenSocket.on('close', function() {
+    //HACK: Wait to exit the process to ensure that the parent
+    //process has had time to receive all messages via process.send()
+    //This may be indicitave of some other issue.
+    setTimeout(function() {
```

Comment:
This might be a bad idea. Hacks like this almost always result in flaky tests in the future. We need a more reliable way to do this.